rust-src/wallet_library/src/wallet.rs

```diff
@@ -0,0 +1,338 @@
+    let key = wallet.get_account_public_key(
+        identity_provider_index,
+        identity_index,
+        credential_counter,
+    )?;
+    Ok(base16_encode_string(&key))
+}
+
+/// Get the hex encoded PRF-key for the seed `seed_as_hex`, the net `net`,
+/// identity provider `identity_provider_index` and identity `identity_index`.
+pub fn get_prf_key_aux(
+    seed_as_hex: HexString,
+    net_as_str: &str,
+    identity_provider_index: u32,
+    identity_index: u32,
+) -> Result<HexString> {
+    let net = get_net(net_as_str)?;
+    let wallet = get_wallet(seed_as_hex, net)?;
+    let key = wallet.get_prf_key(identity_provider_index, identity_index)?;
+    Ok(base16_encode_string(&key))
+}
+
+/// Get the hex encoded IdCredSec for the seed `seed_as_hex`, the net `net`,
+/// identity provider `identity_provider_index` and identity `identity_index`.
+pub fn get_id_cred_sec_aux(
+    seed_as_hex: HexString,
+    net_as_str: &str,
+    identity_provider_index: u32,
+    identity_index: u32,
+) -> Result<HexString> {
+    let net = get_net(net_as_str)?;
+    let wallet = get_wallet(seed_as_hex, net)?;
+    let key = wallet.get_id_cred_sec(identity_provider_index, identity_index)?;
+    Ok(base16_encode_string(&key))
+}
+
+/// Get the hex encoded signature blinding randomness for the seed
+/// `seed_as_hex`, the net `net`, identity provider `identity_provider_index`
+/// and identity `identity_index`.
+pub fn get_signature_blinding_randomness_aux(
+    seed_as_hex: HexString,
+    net_as_str: &str,
+    identity_provider_index: u32,
+    identity_index: u32,
+) -> Result<HexString> {
+    let net: Net = get_net(net_as_str)?;
+    let wallet = get_wallet(seed_as_hex, net)?;
+    let key = wallet.get_blinding_randomness(identity_provider_index, identity_index)?;
+    Ok(base16_encode_string(&key))
+}
+
+/// Get the hex encoded attribute commitment randomness for the seed
+/// `seed_as_hex`, the net `net`, identity provider `identity_provider_index`,
+/// identity `identity_index`, credential counter `credential_counter` and the
+/// provided attribute `attribute`.
+pub fn get_attribute_commitment_randomness_aux(
+    seed_as_hex: HexString,
+    net_as_str: &str,
+    identity_provider_index: u32,
+    identity_index: u32,
+    credential_counter: u32,
+    attribute: u8,
+) -> Result<HexString> {
+    let net: Net = get_net(net_as_str)?;
+    let wallet = get_wallet(seed_as_hex, net)?;
+    let key = wallet.get_attribute_commitment_randomness(
+        identity_provider_index,
+        identity_index,
+        credential_counter,
+        AttributeTag(attribute),
+    )?;
+    Ok(base16_encode_string(&key))
+}
+
+/// Get the hex encoded verifiable credential signing key for seed
+/// `seed_as_hex`, net `net`, issuer index `issuer_index`, issuer subindex
+/// `issuer_subindex` and verifiable credential
+/// index `verifiable_credential_index`.
+pub fn get_verifiable_credential_signing_key_aux(
+    seed_as_hex: HexString,
+    net_as_str: &str,
```

Comment:
Is there value in these being `str` instead of just `Net`?

Similarly for HexString, is there value in using that instead of the types we actually want, and relying on their JSON serialization to convert when needed?

Reply:
We decided to leave it as is.